### ISSUE TITLE
mypy doesn't complain about Forward References

### ISSUE THREAD

**Issue by rowillia** [python/mypy#1025]
The following code passes through mypy no problem:

``` py
class Foo:
    def __init__(self):
        self.foo = 123

    def get_me(self) -> Foo:
        return self
```

According to https://www.python.org/dev/peps/pep-0484/#forward-references, this should be written as:

``` py
class Foo:
    def __init__(self):
        self.foo = 123

    def get_me(self) -> 'Foo':
        return self
```

**Comment by JukkaL** [comment]
Yeah, mypy should catch this issue. I think that there may be a duplicate issue about this open already. At least I remember a discussion about this.

**Comment by ilevkivskyi** [comment]
Duplicate of #3088
